ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -1725,3 +1560,199 @@ func TestCalculateStatus(t *testing.T) {
 	assert.Equal(t, headNodeIP, newInstance.Status.Head.PodIP)
 	assert.Equal(t, headServiceIP, newInstance.Status.Head.ServiceIP)
 }
+
+func Test_TerminatedWorkers_NoAutoscaler(t *testing.T) {
+	setupTest(t)
+
+	// TODO (kevin85421): The tests in this file are not independent. As a workaround,
+	// I added the assertion to prevent the test logic from being affected by other changes.
+	// However, we should refactor the tests in the future.
+
+	// This test makes some assumptions about the testRayCluster object.
+	// (1) 1 workerGroup
+	// (2) The goal state of the workerGroup is 3 replicas.
+	// (3) Set the `WorkersToDelete` field to an empty slice.
+	// (4) Disable autoscaling.
```

Comment:
Another basic question, it seems like even though autoscaling is disabled, we're scaling nodes up and down to meet the goal state.  So what does "autoscaling" do? (Is it the part that sets the goal state based on Ray's resource demands?)

Reply:
This test simulates disabling Autoscaler by making `WorkersToDelete` empty, so we can make sure the Pods are deleted by the terminated state checks rather than `WorkersToDelete`.